manual/manual/refman/exten.etex

```diff
@@ -1546,16 +1568,31 @@ module X = struct
   [@@@warning "+9"]  (* locally enable warning 9 in this structure *)
   ...
 end
-  [@@deprecated "Please is module 'Y' instead."]
+  [@@deprecated "Please use module 'Y' instead."]
 
 let x = begin[@warning "+9] ... end in ....
 
 type t = A | B
   [@@deprecated "Please use type 's' instead.]
 
-
 let f x =
   assert (x >= 0) [@ppwarning "TODO: remove this later"];
+
+let rec no_op = function
+  | [] -> ()
+  | _ :: q -> (no_op[@tailcall]) q;;
+
+let f x = x [@@inline]
+
+let () = (f[@inlined]) ()
+
+type fragile =
+  | Int of int [@warn_on_literal_pattern]
+  | String of string [@warn_on_literal_pattern]
+
+let f = function
+| Int 0 | String "constant" -> () (* trigger warning *)
```

Comment:
Warning 52 in fact. I have added the warning number, and fixed it in the attribute description.